README.md

```diff
@@ -245,55 +246,63 @@ You supply either an IAM service **API key** or an **access token**:
 Supplying the IAM API key:
 
 ```java
-// in the constructor, letting the SDK manage the IAM token
+// letting the SDK manage the IAM token
 IamOptions options = new IamOptions.Builder()
   .apiKey("<iam_api_key>")
   .url("<iam_url>") // optional - the default value is https://iam.cloud.ibm.com/identity/token
   .build();
 Discovery service = new Discovery("2017-11-07", options);
 ```
 
-```java
-// after instantiation, letting the SDK manage the IAM token
-Discovery service = new Discovery("2017-11-07");
-IamOptions options = new IamOptions.Builder()
-  .apiKey("<iam_api_key>")
-  .build();
-service.setIamCredentials(options);
-```
-
 Supplying the access token:
 
 ```java
-// in the constructor, assuming control of managing IAM token
+// assuming control of managing IAM token
 IamOptions options = new IamOptions.Builder()
   .accessToken("<access_token>")
   .build();
 Discovery service = new Discovery("2017-11-07", options);
 ```
 
+#### Username and password
+
 ```java
-// after instantiation, assuming control of managing IAM token
-Discovery service = new Discovery("2017-11-07");
-IamOptions options = new IamOptions.Builder()
-  .accessToken("<access_token>")
+// in the constructor
+BasicAuthConfig config = new BasicAuthConfig.Builder()
+  .username("<username>")
+  .password("<password")
   .build();
-service.setIamCredentials(options);
+Discovery service = new Discovery("2017-11-07", config);
 ```
 
-#### Username and password
+#### ICP
+Like IAM, you can pass in credentials to let the SDK manage an access token for you or directly supply an access token to do it yourself.
 
 ```java
-// in the constructor
-Discovery service = new Discovery("2017-11-07", "<username>", "<password>");
+// letting the SDK manage the token
+ICP4DConfig config = new ICP4DConfig.Builder()
+  .url("<ICP token exchange base URL>")
+  .username("<username>")
+  .password("<password>")
+  .disableSSLVerification(true)
+  .build();
+Discovery service = new Discovery("2017-11-07", config);
+service.setEndPoint("<service ICP URL>");
 ```
 
 ```java
-// after instantiation
-Discovery service = new Discovery("2017-11-07");
-service.setUsernameAndPassword("<username>", "<password>");
+// letting the SDK manage the token
```

Comment:
Should this be `// user managed token` or something similar?

Reply:
Yeah maybe for clarity. Good call 👍

Reply:
Ohh yep, copy and paste error 😶  I'll fix that real quick.